drink/src/runtime/pallet_contracts_debugging.rs

```diff
@@ -0,0 +1,114 @@
+//! This module provides all the necessary elements for supporting contract debugging directly in
+//! the contracts pallet.
+//!
+//! # Smart-contract developer <-> pallet-contracts interaction flow
+//!
+//! The interaction between end-user and runtime is as follows:
+//! 1. At some points during execution, the pallet invokes some callback through its configuration
+//! parameter `Debug`.
+//! 2. In order to forward the callback outside the runtime, `Debug` will call a runtime interface,
+//! that will then forward the call further to the proper runtime extension.
+//! 3. The runtime extension can be fully controlled by the end-user. It just has to be registered
+//! in the runtime.
+//!
+//! So, in brief: pallet-contracts -> runtime interface -> runtime extension
+//!              |<-----------runtime side-------------->|<---user side--->|
+//!
+//! # Passing objects between runtime and runtime extension
+//!
+//! Unfortunately, runtime interface that lies between runtime and the end-user accepts only
+//! very simlpe argument types and those that implement some specific traits. This means that
```

Comment:
```suggestion
//! very simple argument types and those that implement some specific traits. This means that
```

Reply:
meh, my IDE stopped spellchecking for some reason :| fixed